test/regex.jl

```diff
@@ -61,3 +61,6 @@ end
 # 'a' flag to disable UCP
 @test match(r"\w+", "Düsseldorf").match == "Düsseldorf"
 @test match(r"\w+"a, "Düsseldorf").match == "D"
+
+@test startswith("abc", r"a")
+@test endswith("abc", r"c")
```

Comment:
This should also test patterns that would match but don't match at the start or the end, e.g.
```jl
@test !startswith("abc", r"b")
@test !startswith("abc", r"c")
@test !endswith("abc", r"a")
@test !endswith("abc", r"b")
```

Reply:
Thanks!  Added those and a few extra.